library/std/src/sys/windows/io.rs

```diff
@@ -78,3 +82,120 @@ impl<'a> IoSliceMut<'a> {
         unsafe { slice::from_raw_parts_mut(self.vec.buf as *mut u8, self.vec.len as usize) }
     }
 }
+
+#[unstable(feature = "is_terminal", issue = "80937")]
+impl io::IsTerminal for sys::stdio::Stdin {
+    fn is_terminal() -> bool {
```

Comment:
There's a lot of duplication here, this should be factored out into a method that takes `fd` and `others` as arguments.